client/html/src/Client/Html/Catalog/Product/Standard.php

```diff
@@ -0,0 +1,468 @@
+		 *
+		 * Decorators extend the functionality of a class by adding new aspects
+		 * (e.g. log what is currently done), executing the methods of the underlying
+		 * class only in certain conditions (e.g. only for logged in users) or
+		 * modify what is returned to the caller.
+		 *
+		 * This option allows you to wrap local decorators
+		 * ("\Aimeos\Client\Html\Catalog\Decorator\*") around the html client.
+		 *
+		 *  client/html/catalog/product/decorators/local = array( 'decorator2' )
+		 *
+		 * This would add the decorator named "decorator2" defined by
+		 * "\Aimeos\Client\Html\Catalog\Decorator\Decorator2" only to the html client.
+		 *
+		 * @param array List of decorator names
+		 * @since 2019.06
+		 * @category Developer
+		 * @see client/html/common/decorators/default
+		 * @see client/html/catalog/product/decorators/excludes
+		 * @see client/html/catalog/product/decorators/global
+		 */
+
+		return $this->createSubClient( 'catalog/product/' . $type, $name );
+	}
+
+
+	/**
+	 * Processes the input, e.g. store given values.
+	 * A view must be available and this method doesn't generate any output
+	 * besides setting view variables.
+	 */
+	public function process()
+	{
+		$context = $this->getContext();
+		$view = $this->getView();
+
+		try {
+			parent::process();
+		} catch (\Aimeos\Client\Html\Exception $e) {
+			$error = array($context->getI18n()->dt( 'client', $e->getMessage() ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+		} catch (\Aimeos\Controller\Frontend\Exception $e) {
+			$error = array($context->getI18n()->dt( 'controller/frontend', $e->getMessage() ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+		} catch (\Aimeos\MShop\Exception $e) {
+			$error = array($context->getI18n()->dt( 'mshop', $e->getMessage() ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+		} catch (\Exception $e) {
+			$error = array($context->getI18n()->dt( 'client', 'A non-recoverable error occured' ));
+			$view->productErrorList = $view->get( 'productErrorList', [] ) + $error;
+			$this->logException( $e );
+		}
+	}
+
+
+	/**
+	 * Returns the list of sub-client names configured for the client.
+	 *
+	 * @return array List of HTML client names
+	 */
+	protected function getSubClientNames()
+	{
+		return $this->getContext()->getConfig()->get( $this->subPartPath, $this->subPartNames );
+	}
+
+
+	/**
+	 * Modifies the cached body content to replace content based on sessions or cookies.
+	 *
+	 * @param string $content Cached content
+	 * @param string $uid Unique identifier for the output if the content is placed more than once on the same page
+	 * @return string Modified body content
+	 */
+	public function modifyBody( $content, $uid )
+	{
+		$content = parent::modifyBody( $content, $uid );
+
+		return $this->replaceSection( $content, $this->getView()->csrf()->formfield(), 'catalog.lists.items.csrf' );
```

Comment:
You should use `catalog.product.csrf` here

Reply:
Then I can no longer use the common product partial.

https://github.com/aimeos/ai-client-html/blob/e59e2d7f8e0ff2fa6269261557313762fe2d7048/client/html/templates/common/partials/products-standard.php#L262-L264

Reply:
OK, good point